### PR TITLE
Execute legs in order

### PR DESCRIPTION
**Pull request by maxsam4**
Added an unrelated test case, plus:

### modifed logic
- Legs of a settlement instruction are now executed in order.

### CODE REVIEW

**Review by satyamakgec — APPROVED** [review]
LGTM, If possible can we add a test case for 0 price STO

**Comment by satyamakgec** [comment]
ignore the comment, I found the related test case in another PR of yours.

**Review by CJP10 — APPROVED** [review]
LGTM

File: pallets/settlement/src/lib.rs [hunk 1]
```diff
@@ -990,7 +990,9 @@ impl<T: Trait> Module<T> {
     }
 
     fn execute_instruction(instruction_id: u64) -> Result<u32, DispatchError> {
-        let legs = <InstructionLegs<T>>::iter_prefix(instruction_id).collect::<Vec<_>>();
+        let mut legs = <InstructionLegs<T>>::iter_prefix(instruction_id).collect::<Vec<_>>();
+        // NB: Execution order doesn't matter in most cases but might matter in some edge cases around compliance
```
Comment (Centril) [inline pallets/settlement/src/lib.rs:994]: Could you add an example of an edge case here for future readers?

**Review by fmiguelgarcia — APPROVED** [review]

**Review by Centril — APPROVED** [review]

**Pull request merged**